src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,129 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
                 if (!targetEntityGraphQLDetails.Enabled)
                 {
                     HandleOrRecordException(new DataApiBuilderException(
-                        message: $"entity: {relationship.TargetEntity} is disabled for GraphQL.",
+                        message: $"Entity: {relationship.TargetEntity} is disabled for GraphQL.",
                         statusCode: HttpStatusCode.ServiceUnavailable,
                         subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
                 }
 
+                // Validation to ensure that if source fields exist, target fields exist as well.
+                if (relationship.SourceFields is not null && relationship.TargetFields is null)
```

Comment:
Nit/suggestion: The number of && conditionals in the IF statements can be significantly reduced if we start with the condition:

`if (relationship.SourceFields is not null || relationship.TargetFields is not null)`, because then only we need to do the validations. All the other cases will go inside this if condition. If this condition is returns false, we don't need to do any validations at all.

Reply:
It is a good suggestion, and I considered some other options, but I think this way is most readable, and the conditionals end up describing the cases well.